ts/Data/Stores/DataStore.ts

```diff
@@ -49,6 +50,41 @@ class DataStore {
     *
     * */
 
+    protected getMetadataJSON(): DataStore.MetadataJSON {
+        const json = [];
+        let elem;
+
+        for (let i = 0, iEnd = this.metadata.length; i < iEnd; i++) {
+            elem = this.metadata[i];
+
+            json.push([
+                elem.name,
+                elem.metadata
+            ]);
+        }
+
+        return json;
+    }
+
+    protected getMetadataFromJSON(metadataJSON: DataStore.MetadataJSON):
+    Array<DataStore.MetaColumn> {
+        const metadata = [];
+        let elem;
+
+        for (let i = 0, iEnd = metadataJSON.length; i < iEnd; i++) {
+            elem = metadataJSON[i];
+
+            if (elem instanceof Array && typeof elem[0] === 'string') {
+                metadata.push({
+                    name: elem[0],
+                    metadata: elem[1]
+                });
+            }
+        }
+
+        return metadata;
+    }
```

Comment:
Thanks! This looks good. One thing: This should become static and moved to the top of the class, as it is no requiring an instance.